pyaedt/q3d.py

```diff
@@ -368,6 +370,355 @@ def edit_sources(
+                            matrix_type_array = matrix_type.split(", ")
+                            if [x for x in matrix_type_array if x == "Spice"]:
+                                self.logger.error("Spice can't be a matrix type if problem types are AC RL or DC RL")
+                                return False
+                else:
+                    self.logger.error("Invalid problem type. Available values are C, AC RL, DC RL")
+                    return False
+
+        if variations is None:
+            if not self.available_variations.nominal_w_values_dict:
+                variations = ""
+            else:
+                variations_list = []
+                for x in range(0, len(self.available_variations.nominal_w_values_dict)):
+                    variation = "{}='{}'".format(
+                        list(self.available_variations.nominal_w_values_dict.keys())[x],
+                        list(self.available_variations.nominal_w_values_dict.values())[x],
+                    )
+                    variations_list.append(variation)
+                variations = ",".join(variations_list)
+
+        if setup_name is None:
+            setup_name = self.analysis_setup
+        elif setup_name != self.analysis_setup:
+            self.logger.error("Setup name is not valid, please provide a valid analysis setup name.")
+            return False
+        if sweep is None:
+            sweep = self.design_solutions.default_adaptive
+        else:
+            sweep_array = [x.split(": ")[1] for x in self.existing_analysis_sweeps]
+            if sweep.replace(" ", "") not in sweep_array:
+                self.logger.error("Sweep is not valid, please provide a valid sweep.")
+                return False
+        analysis_setup = setup_name + " : " + sweep.replace(" ", "")
+
+        if reduce_matrix is None:
+            reduce_matrix = "Original"
+        else:
+            if self.matrices:
+                if not [matrix for matrix in self.matrices if matrix.name == reduce_matrix]:
+                    self.logger.error("Matrix doesn't exist, provide an existing matrix.")
+                    return False
+            else:
+                self.logger.error("Matrix list parameters is empty, can't export a valid matrix.")
+                return False
+
+        if r_unit is None:
+            r_unit = "ohm"
+        else:
+            if not r_unit.endswith("ohm"):
+                self.logger.error("Provide a valid unit for resistor.")
+                return False
+
+        if l_unit is None:
+            l_unit = "nH"
+        else:
+            if not l_unit.endswith("H"):
+                self.logger.error("Provide a valid unit for inductor.")
+                return False
+
+        if c_unit is None:
+            c_unit = "pF"
+        else:
+            if c_unit not in ["fF", "pF", "nF", "uF", "mF", "farad"]:
+                self.logger.error("Provide a valid unit for capacitance.")
+                return False
```

Comment:
No need to set the default value of `c_unit` to `None`.
```suggestion
        if c_unit not in ["fF", "pF", "nF", "uF", "mF", "farad"]:
            self.logger.error("Provide a valid unit for capacitance.")
            return False
```